src/transformers/generation/utils.py

```diff
@@ -4484,18 +4485,18 @@ def assisted_decoding(
             # 2.2. Process the new logits
             new_logits = outputs.logits[:, -candidate_length - 1 :]  # excludes the input prompt if present
             if len(logits_processor) > 0:
-                for i in range(candidate_length):
+                for i in range(candidate_length + 1):
```

Comment:
That was a bug previously. We forgot to apply the logits processors to the last logit here

Reply:
Good catch 👀